src/status_im/contexts/wallet/router/core.cljs

```diff
@@ -0,0 +1,87 @@
+(ns status-im.contexts.wallet.router.core
+  (:require [schema.core :as schema]
+            [status-im.contexts.wallet.router.schema :as router.schema]
+            [status-im.contexts.wallet.send.transaction-settings.core :as transaction-settings]
+            [status-im.contexts.wallet.send.utils :as send-utils]))
+
+(def ^:private gwei-precision 6)
+
+(defn- to-gwei
+  [amount]
+  (send-utils/convert-to-gwei amount gwei-precision))
+
+(defn transaction-fees-by-mode
+  [route]
+  (let [{:keys [low medium high low-estimated-time
+                medium-estimated-time high-estimated-time]}
+        (:suggested-levels-for-max-fees-per-gas route)]
+    {:tx-fee-mode/normal
+     {:max-fees       (to-gwei low)
+      :estimated-time low-estimated-time}
+
+     :tx-fee-mode/fast
+     {:max-fees       (to-gwei medium)
+      :estimated-time medium-estimated-time}
+
+     :tx-fee-mode/urgent
+     {:max-fees       (to-gwei high)
+      :estimated-time high-estimated-time}}))
+
+(schema/=> transaction-fees-by-mode
+  [:=>
+   [:cat router.schema/?route]
+   [:map-of
+    router.schema/?fee-modes
+    [:map {:closed? true}
+     [:max-fees :string]
+     [:estimated-time :int]]]])
+
+(defn transaction-fee-mode
+  [route]
+  (-> route
+      :tx-gas-fee-mode
+      transaction-settings/gas-rate->tx-fee-mode))
+
+(schema/=> transaction-fee-mode
+  [:=>
+   [:cat router.schema/?route]
+   router.schema/?fee-modes])
+
+(defn transaction-estimated-time
+  "Get the transaction estimated time. If the time is unknown (0),
+  fallback to the current fee-mode's estimated time."
+  [route]
+  (let [estimated-time (:tx-estimated-time route)]
+    (if (zero? estimated-time)
+      (-> route
+          transaction-fees-by-mode
+          (get (transaction-fee-mode route))
+          :estimated-time)
+      estimated-time)))
+
+(schema/=> transaction-estimated-time
+  [:=>
+   [:cat router.schema/?route]
+   :int])
+
+(defn transaction-gas-fees
+  [route]
+  (let [{:keys [tx-base-fee tx-priority-fee
+                tx-l-1-fee tx-max-fees-per-gas]} route]
+    {:gas-price                "0"
+     :eip-1559-enabled         true
+     :base-fee                 (to-gwei tx-base-fee)
+     :max-priority-fee-per-gas (to-gwei tx-priority-fee)
+     :l-1-gas-fee              (to-gwei tx-l-1-fee)
+     :tx-max-fees-per-gas      (to-gwei tx-max-fees-per-gas)}))
+
+(schema/=> transaction-gas-fees
+  [:=>
+   [:cat router.schema/?route]
+   [:map {:closed? true}
```

Comment:
Why close the schema if the data is coming from status-go? If the producer adds keys, the consumer will break (fail instrumentation).

Reply:
this is the schema for the return value, so IMO it should be closed so it fails in case we add new fields on the client side.

Reply:
Good call 👍🏼